doc/man3/OSSL_CMP_CTX_new.pod

```diff
@@ -0,0 +1,654 @@
+It may make use of a custom defined argument stored in the ctx
+by means of OSSL_CMP_CTX_set_http_cb_arg(),
+which may be retrieved again through OSSL_CMP_CTX_get_http_cb_arg().
+
+OSSL_CMP_CTX_set_http_cb_arg() sets an argument, respectively a pointer to
+a structure containing arguments,
+optionally to be used by the http connect/disconnect callback function.
+B<arg> is not consumed, and it must therefore explicitly be freed when not
+needed any more. B<arg> may be NULL to clear the entry.
+
+OSSL_CMP_CTX_get_http_cb_arg() gets the argument, respectively the pointer to a
+structure containing arguments, previously set by
+OSSL_CMP_CTX_set_http_cb_arg().
+
+OSSL_CMP_CTX_set_transfer_cb() sets the message transfer callback function,
+which has the type
+
+ typedef int (*OSSL_cmp_transfer_cb_t)(const OSSL_CMP_CTX *ctx,
+                                       const OSSL_CMP_MSG *req,
+                                       OSSL_CMP_MSG **res);
+Returns 1 on success, 0 on error.
+
+Default is OSSL_CMP_MSG_http_perform().
+The callback should send the CMP request it obtains via the B<req> parameter
+and on success place the response in the B<*res> output parameter.
+The transfer callback may make use of a custom defined argument stored in
+the ctx by means of OSSL_CMP_CTX_set_transfer_cb_arg(), which may be retrieved
+again through OSSL_CMP_CTX_get_transfer_cb_arg().
+On success the cb must return 0, else a CMP error reason code defined in cmp.h.
+
+
+OSSL_CMP_CTX_set_transfer_cb_arg() sets an argument, respectively a pointer to a
+structure containing arguments, optionally to be used by the transfer callback.
+B<arg> is not consumed, and it must therefore explicitly be freed when not
+needed any more. B<arg> may be NULL to clear the entry.
+
+OSSL_CMP_CTX_get_transfer_cb_arg() gets the argument, respectively the pointer
+to a structure containing arguments, previously set by
+OSSL_CMP_CTX_set_transfer_cb_arg().
+
+OSSL_CMP_CTX_set1_srvCert() pins the server certificate to be directly trusted
+(even if it is expired) for verifying response messages.
+The cert pointer is not consumed. It may be NULL to clear the entry.
+
+OSSL_CMP_CTX_set1_expected_sender() sets the Distinguished Name (DN) expected to
+be given in the sender response for messages protected with MSG_SIG_ALG. This
+may be used to enforce that during validation of received messages the given DN
+matches the sender field of the PKIMessage header, which in turn is used to
+identify the server certificate.
+This can be used to ensure that only a particular entity is accepted to act as
+CMP server, and attackers are not able to use arbitrary certificates of a
+trusted PKI hierarchy to fraudulently pose as server.
+This defaults to the subject DN of the certificate set via
+OSSL_CMP_CTX_set1_srvCert(), if any.
+
+OSSL_CMP_CTX_set0_trustedStore() sets the X509_STORE type certificate store
+containing trusted (root) CA certificates and possibly CRLs and a certificate
```

Comment:
and ... and ... 
Second time, i read this phrase construction

Reply:
I split the phrase into two sentences. Hope it will be more clear now.